src/midi-player.ts

```diff
@@ -16,31 +17,117 @@ export class MidiPlayer implements IMidiPlayer {
 
     private _offset: null | number;
 
+    private _paused: null | number;
+
+    private _latest: null | number;
+
     private _resolve: null | (() => void);
 
     private _scheduler: Scheduler;
 
     private _schedulerSubscription: null | { unsubscribe(): void };
 
+    private _channels: null | number[];
+
     constructor({ encodeMidiMessage, json, midiFileSlicer, midiOutput, scheduler }: IMidiPlayerOptions) {
         this._encodeMidiMessage = encodeMidiMessage;
         this._endedTracks = null;
         this._json = json;
         this._midiFileSlicer = midiFileSlicer;
         this._midiOutput = midiOutput;
         this._offset = null;
+        this._paused = null;
+        this._latest = null;
         this._resolve = null;
         this._scheduler = scheduler;
         this._schedulerSubscription = null;
+
+        // List all channels that are contained in the MIDI file.
+        // We will use it to send All Sound Off messages on pause / stop.
+        this._channels = [...new Set(
+            this._json.tracks.reduce((channels, track) => {
+                const event = track.find(event => 'channel' in event);
+                if (event) {
+                    channels.push((event as IMidiStatusEvent).channel)
+                }
+                return channels;
+            }, [] as number[])
+        )];
```

Comment:
Is it safe to assume that each track only contains events for one channel?

I think we can store the channels as a `Set`.

```suggestion
    this._channels = this._json.tracks.reduce((channels, track) => {
        const event = track.find((event: IMidiMetaEvent | IMidiStatusEvent): event is IMidiStatusEvent => 'channel' in event);
            
        if (event === undefined) {
            return channels;
        }
             
        return channels.add(event.channel);
    }, new Set());
```

Reply:
> Is it safe to assume that each track only contains events for one channel?

Actually, you're right, it's not safe. I'll rework this part to turn `channels` into a set, but to read all the events containing `channel`.

Reply:
Done